mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,497 @@ source include/wait_show_condition.inc;
+show status like 'rpl_semi_sync_master_get_ack';
+
+--echo # Test 10: Testing `rpl_semi_sync_master_timeout=0`
+--echo # Note 1:  For asynchronous replication `Gtid_state_sent` is not null,
+--echo #          but `Gtid_state_ack` is null.
+--echo # Note 2:  We want that `rpl_semi_sync_master_timeout=0` be edge case,
+--echo #          by mimicking asynchronous replication, as semi-sync stalled
+--echo #          replication type, for which `Gtid_state_ack` should not update
+--echo #          or be null (see Note 3). Also `rpl_semi_sync_master_status` has
+--echo #          to remain `ON`, instead switching off, so instead of
+--echo #          no waiting on ACK, primary must wait on ACK from replica.
+--echo # Note 3:  With this option we want `Gtid_state_ack` to reflect
+--echo #          the latest received transaction by the replica[s].
+--echo # -----------------------------------------------------------------------
+# Test DBUG_SYNC by stopping the slave and testing the new sync point
+# Note we need to stop both slaves in order to control sent ACK for new active transaction
+
+--let $save_default_master_timeout= `SELECT @@GLOBAL.rpl_semi_sync_master_timeout`
+select @@global.rpl_semi_sync_master_timeout;
+show variables like 'rpl_semi_sync_master_enabled';
+
+# Master status must be `ON`
+let $status_var= rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_master_status';
+
+show status like 'rpl_semi_sync_master_get_ack';
+set global rpl_semi_sync_master_timeout=0;
+select @@global.rpl_semi_sync_master_timeout;
+# here rpl_semi_sync_master_status=0 means async is taking place (no wait on ACK from slave)
+show variables like 'rpl_semi_sync_master_timeout';
+show status like 'rpl_semi_sync_master_status';
+
+connection slave;
+--source include/stop_slave.inc
+--let $save_server_2_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+SET debug_sync='now WAIT_FOR at_slave_reply TIMEOUT 10';
+
+connection slave2;
+--source include/stop_slave.inc
+--let $save_server_3_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+SET debug_sync='now WAIT_FOR at_slave_reply TIMEOUT 10';
+
+--echo # Both replicas are stopped with synchronization point
+connection master;
+# Write event
+INSERT INTO t values (1111);
+# There will be no ACK so good enough to check if the same rpl_semi_sync_master_get_ack is observed
+let $status_var= rpl_semi_sync_master_get_ack;
+let $status_var_value= 9;
```

Comment:
A couple notes on testing the `_get_ack` value...
1. It'd be better to use a dynamic value here, i.e. save the value initially at the beginning of the test case (alongside line 431), like
```
let $acks_initial= query_get_value(SHOW STATUS LIKE 'rpl_semi_sync_master_get_ack', Value, 1);
```

2. This value should not increase from the initial (as neither replica should have ACK'd yet, as both should be stopped at the debug_sync point); so it should be `$acks_initial`. Then as your `now SIGNAL reply_ack_to_master`; then it should go up one at a time (dynamically by using `SELECT ($acks_initial + 1)`, for example)